File: CS/BandIcon/Form1.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Text;
using System.Windows.Forms;
using DevExpress.XtraReports.UI;

namespace BandIcon
{
    public partial class Form1 : Form
    {
        public Form1()
        {
            InitializeComponent();
        }

        private void Form1_Load(object sender, EventArgs e)
        {
            XtraReport report = new XtraReport();
            MyBand band = new MyBand();
            report.Bands.Add(band);
            report.ShowDesignerDialog();
            this.Close();
        }
    }
    [ToolboxBitmap("XRDetailBand.gif")]
    class MyBand : DetailBand
    {
    }
}

File: VB/BandIcon/Form1.vb
Imports Microsoft.VisualBasic
Imports System
Imports System.Collections.Generic
Imports System.ComponentModel
Imports System.Data
Imports System.Drawing
Imports System.Text
Imports System.Windows.Forms
Imports DevExpress.XtraReports.UI

Namespace BandIcon
	Partial Public Class Form1
		Inherits Form
		Public Sub New()
			InitializeComponent()
		End Sub

		Private Sub Form1_Load(ByVal sender As Object, ByVal e As EventArgs) Handles MyBase.Load
			Dim report As New XtraReport()
			Dim band As New MyBand()
			report.Bands.Add(band)
			report.ShowDesignerDialog()
			Me.Close()
		End Sub
	End Class
	<ToolboxBitmap("XRDetailBand.gif")> _
	Friend Class MyBand
		Inherits DetailBand
	End Class
End Namespace